processor/src/test/java/org/mapstruct/ap/test/bugs/_1395/NotUsedService.java

```diff
@@ -0,0 +1,25 @@
+/**
+ *  Copyright 2012-2017 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.test.bugs._1395;
+
+/**
+ * @author Filip Hrisafov
+ */
+public interface NotUsedService {
```

Comment:
Shouldn't there be an `@Mapper`?

Reply:
Not really. One doesn't have to use mappers in `uses`. You can use any class / service in there

Reply:
I know.. But the behaviour is different. For the 'normal' component mapper, MapStruct initialises the instance variable with `Mappers.getMapper(... )` instead of `new ...`. Filtering is carried out if the mapper is indeed used. When not, there's not even going to be a field. 

So, to make a long story short it could be that the code above is not even triggered because MapStruct already decided there's no need. Or perhaps.. I'm just seeing stuff.  :smile:

Reply:
> But the behaviour is different. For the 'normal' component mapper, MapStruct initialises the instance variable with Mappers.getMapper(... ) instead of new .... 

Not when an injection `componentModel` is used (`spring`, `jsr330`, `cdi`). When one of the non defaults is used, then the uses mappers or services are always injected.

>So, to make a long story short it could be that the code above is not even triggered because MapStruct already decided there's no need. Or perhaps.. I'm just seeing stuff.

That is the entire idea of the PR. To show that the `NotUsedService` should not be injected in the mapper 😄.